### DISCUSSION TITLE
Resizing and column width

### DISCUSSION THREAD

**Discussion by sylcha** [quasarframework/quasar#12635] [General - Components / Directives / etc]
Hi everyone !
I'm working on an app using quasar framework and p5.js library. I insert my p5 canvas into a `div`  with class `col`. The p5 canvas take its width directly from this `div` element using its `id` attribute.
```html
<template>
  <q-page padding class="bg-primary">
    <div class="column justify-center" style="min-height: inherit">
      <div class="row justify-center">
        <div id="canvasPlace" class="col">
          <div id="p5Canvas"></div>
        </div>
      </div>
      <div class="row justify-center text-white">
        <div class="col-1">
          <q-btn flat dense round icon="ti-arrow-left" aria-label="Gauche" />
        </div>
        <div class="col-1">
          <q-btn flat dense round icon="ti-reload" aria-label="Retour" />
        </div>
        <div class="col-1">
          <q-btn flat dense round icon="ti-arrow-right" aria-label="Droite" />
        </div>
      </div>
    </div>
  </q-page>
</template>

<script>
const P5 = require("p5");
const dessin = require("../js/ZoneNombre.js");
let d;
export default {
  name: "Nombres",
  mounted() {
    // NOTE: Use p5 as an instance mode
    d = new P5(dessin.main, "p5Canvas");
  },
};
</script>
```

In my p5 sketch, I use the `windowResized` function to set up the width of the sketch according to the width of the canvas.

```js
p5.windowResized = () => {
    let container = document.getElementById("canvasPlace");
    console.log("width =" + container.clientWidth);
    p5.resizeCanvas(container.clientWidth, canvasHeight);
  };
```

When I increase the size of the window, it works fine : the column grows and the canvas grows accordingly. However, when I shrink the window, the column width does not decrease. In fact, its width does not change.

I think it is because when I shrink the window, the column already hosts the canvas and needs to be as large as the canvas is : then, the column cannot shrink and the canvas does not adjust since the width of the column does not change.

I cannot figure out a solution to this problem : I would like that the column width decreases (and the canvas width consequently) as I shrink the window. That's why I am asking some help here. Thanks in advance !

Sylvain

**Comment by MilosPaunovic** [comment]
Have you tested column increasing/decreasing without canvas element? If that works as expected, I would advise you to post your question to the repository of `p5` - you'll get better support that way.

**Reply by sylcha** [reply]
When I comment the line where the `p5` sketch is created, the page behaves as expected. So my guess was right...

But I've found a workaround: before resizing the canvas based on the column width, I shrink it to 100 pixels. Therefore, the column does not have to host a big canvas and its width can decrease as the window shrinks... It's gross, but it's working

```js
p5.windowResized = () => {
    p5.resizeCanvas(100, canvasHeight);
    let container = document.getElementById("canvasPlace");
    p5.resizeCanvas(container.clientWidth, canvasHeight);
  };
```
